include/boost/beast/core/string.hpp

```diff
@@ -10,21 +10,45 @@
 #ifndef BOOST_BEAST_STRING_HPP
 #define BOOST_BEAST_STRING_HPP
 
+#include <boost/asio/detail/string_view.hpp>
+
+#ifndef BOOST_ASIO_NO_STRING_VIEW
+# if defined(BOOST_ASIO_HAS_STD_STRING_VIEW)
+#  define BOOST_ASIO_HAS_STRING_VIEW 0
+# else
+#  define BOOST_ASIO_HAS_STRING_VIEW 1
+# endif
+#endif
+
 #include <boost/beast/core/detail/config.hpp>
 #include <boost/version.hpp>
+
+#if BOOST_ASIO_NO_STRING_VIEW
 #include <boost/utility/string_view.hpp>
+#endif
```

Comment:
I think we need @chriskohlhoff to weigh in on this, to decide if he wants to support other libraries piggy-backing on the string view declaration.